File: 8kyu/(8 kyu) Area or Perimeter/(8 kyu) Area or Perimeter.js
// 1 Plain Solution
const areaOrPerimeter = function (l, w) {
    const isSquare = l === w;

    if (isSquare) {
        return l * w;
    } else {
        return l * 2 + w * 2;
    }
};
// 2 Optimized solution
const areaOrPerimeter = function (l, w) {
    return l === w
        ? l * w
        : l * 2 + w * 2;
};
// 3 Coding golf
const areaOrPerimeter = (l, w) =>
    l === w
        ? l * w
        : 2 * (l + w);

File: 8kyu/(8 kyu) Area or Perimeter/(8 kyu) Area or Perimeter.java
// 1 Plain solution
public class Solution {
    public static int areaOrPerimeter(int l, int w) {
        final boolean isSquare = l == w;

        if (isSquare) {
            return l * w;
        } else {
            return l * 2 + w * 2;
        }
    }
}

// 2 Optimized solution
public class Solution {
    public static int areaOrPerimeter(int l, int w) {
        return l == w ? l * w : l * 2 + w * 2;
    }
}

// 3 Clever solution
public class Solution {
    public static int areaOrPerimeter(int l, int w) {
        return l == w ? l * w : (l + w) * 2;
    }
}
